flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/typeutils/FieldInfoUtils.java

```diff
@@ -186,6 +186,37 @@ public static boolean isReferenceByPosition(CompositeType<?> ct, Expression[] fi
 		return fieldNames;
 	}
 
+	/**
+	 * Returns field names for a given {@link TypeInformation}. If the input {@link TypeInformation}
+	 * is not a composite type, the result field name should not exist in the invalidNamesForNonCompositeType.
+	 *
+	 * @param inputType The TypeInformation extract the field names.
+	 * @param invalidNamesForNonCompositeType The invalid field names for non-composite types.
+	 * @param <A> The type of the TypeInformation.
+	 * @return An array holding the field names
+	 */
+	public static <A> String[] getFieldNames(TypeInformation<A> inputType, List<String> invalidNamesForNonCompositeType) {
```

Comment:
I think adding a new method with the difference prams is pretty good. but the impl logic can be merged into one method! i.e.:
public static <A> String[] getFieldNames(TypeInformation<A> inputType) {
getFieldNames(inputType, Collections.emptyList());
}
public static <A> String[] getFieldNames(TypeInformation<A> inputType, List<String> existingNames) {
...
}
Whtat do you think?

Reply:
Good idea!